tests/m2m_through/tests.py

```diff
@@ -402,26 +403,45 @@ def test_self_referential_non_symmetrical_clear_first_side(self):
         )
 
     def test_self_referential_symmetrical(self):
-        tony = PersonSelfRefM2M.objects.create(name="Tony")
-        chris = PersonSelfRefM2M.objects.create(name="Chris")
-        Friendship.objects.create(
-            first=tony, second=chris, date_friended=datetime.now()
+        tony = PersonSelfRefM2MSymmetrical.objects.create(name="Tony")
+        chris = PersonSelfRefM2MSymmetrical.objects.create(name="Chris")
+
+        SymmetricalFriendship.objects.create(
+            first=tony, second=chris, date_friended=datetime.now(),
         )
-        Friendship.objects.create(
+
+        self.assertQuerysetEqual(tony.friends.all(), ['Chris'], attrgetter("name"))
+        self.assertQuerysetEqual(chris.friends.all(), [])
+
+        # It's a symmetrical relation but if we create the relationship object by
+        # hand, we need to create both links
+        SymmetricalFriendship.objects.create(
             first=chris, second=tony, date_friended=datetime.now()
         )
 
-        self.assertQuerysetEqual(
-            tony.friends.all(),
-            ['Chris'],
-            attrgetter("name")
-        )
+        self.assertQuerysetEqual(chris.friends.all(), ['Tony'], attrgetter("name"))
 
-        self.assertQuerysetEqual(
-            chris.friends.all(),
-            ['Tony'],
-            attrgetter("name")
-        )
+    def test_self_referential_symmetrical_related_name(self):
+        """
+        Using the related name will get you the link where the object is "first",
+        not the reversed link.
+        """
+        tony = PersonSelfRefM2MSymmetrical.objects.create(name="Tony")
+        chris = PersonSelfRefM2MSymmetrical.objects.create(name="Chris")
+        tony.friends.add(chris, through_defaults={'date_friended': datetime.now()})
+
+        self.assertEqual(
+            tony.symmetricalfriendship_set.first(),
+            SymmetricalFriendship.objects.get(first=tony))
```

Comment:
Can we use custom `date_friended`? and check is it set properly on an intermediary object.